### PR TITLE
Fix Projectiles so they stop hitting source prefab

### PR DESCRIPTION
**Pull request by p3zd1sp3nx0r**
Projectile Script updated to give projectile a live distance. Upon reaching this distance the box collider will go active.
Turret AI updated to give the projectile a live distance
Sample Scene updated to move turrets a bit further away
Bullet Prefab updated to include  default live distance.
Rocket Prefab updated to include default live distance
No changes on helicopter, did not detect any hitting so default of 1 seems to cover

### CODE REVIEW

File: Assets/Scripts/Objects/ProjectileScript.cs [hunk 1]
```diff
@@ -10,34 +11,56 @@ public class ProjectileScript : MonoBehaviour
...
+    private IEnumerator liveCheck()
+    {
+        while (!live)
+        {
+            float _distanceFromSource = Vector3.Distance(InstantiationPosition, transform.position);
+            if (_distanceFromSource > liveDistance)
+            {
+                live = true;
+                this.gameObject.GetComponent<BoxCollider>().enabled = true;
+                StopCoroutine("liveCheck");
```
Comment (ballardi) [inline Assets/Scripts/Objects/ProjectileScript.cs:60]: haven't seen a stopcoroutine called within the method that is the coroutine before. think i assumed stopcoroutine was a thing called based on a decision made by the class rather than the coroutine itself, but if it works it works! (assuming you know this isn't still getting called even after live is false)

File: Assets/Scripts/Objects/ProjectileScript.cs [hunk 2]
```diff
@@ -10,34 +11,56 @@ public class ProjectileScript : MonoBehaviour
...
+            }
+            yield return false;
```
Comment (ballardi) [inline Assets/Scripts/Objects/ProjectileScript.cs:62]: i think this results in a thread calling this method immediately after it returns (so probably even more often that the update method. 
if we wanted to make the coroutine perf better we could reduce the frequency of these calls using an:
            yield return new WaitForSeconds(waitTime);
but at that point it is getting called a similar frequency to even just doing the logic in this method from inside fixedupdate.

In other words i'm thinking the coroutine might be not needed. if we were making projectile live based on /time/ then we could set a coroutine to wait that long and then fire once, which would have been a nice use of coroutine. but since we need to check distance often it's similar to update or fixedupdate i guess.

that said, if it works it works - no need to change it

**Review by ballardi — APPROVED** [review]
